### PR TITLE
Fix sync committee preparation by deduplicating signing requests

### PR DESCRIPTION
**Pull request by hoanhan101**
## Description

This PR fixes an issue in sync committee preparation where duplicate signing requests were being sent to multi-signers. When the same account needs to sign the same root multiple times (which can happen during sync committee duties), the previous implementation would send multiple identical signing requests, potentially causing performance issues or errors in the signing infrastructure.

The fix implements deduplication logic that identifies unique (account, root) pairs before sending signing requests, then maps the results back to the original positions to maintain the expected response structure.

## Changes

- Add deduplication logic in `signRootsMulti` function to identify unique (account, root) pairs
- Maintain mapping between original request positions and deduplicated requests
- Map signatures back to original positions after signing
- Add comprehensive tests for the deduplication logic covering various scenarios including no duplicates, exact duplicates, and mixed patterns

### CODE REVIEW

**Review by Bez625 — CHANGES_REQUESTED** [review]
Overall looks like a good start!

File: services/signer/standard/helpers.go [hunk 1]
```diff
@@ -165,3 +172,54 @@ func (s *Service) signRootsByAccountType(ctx context.Context, accounts []e2wtype
...
+
+// mapSignaturesToOriginalPositions maps unique signatures back to the original request positions.
+func mapSignaturesToOriginalPositions(
+	sigs []phase0.BLSSignature,
+	uniqueSigs []e2types.Signature,
+	dedup deduplicationResult,
+) {
+	for i := range sigs {
+		uniqueIndex := dedup.originalToUniqueIndex[i]
+		if uniqueIndex < len(uniqueSigs) && uniqueSigs[uniqueIndex] != nil {
```
Comment (Bez625) [inline services/signer/standard/helpers.go:221]: If any of these conditions are not met we will silently ignore that signature. Is there a reason for this? If not I think we should return an appropriate error.

**Review by Bez625 — CHANGES_REQUESTED** [review]

**Review by Bez625 — APPROVED** [review]
